src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs

```diff
@@ -110,6 +117,42 @@ internal void AppendSwitchAliased(string switchName, string alias, string parame
             AppendTextWithQuoting(parameter);
         }
 
+        /// <summary>
+        /// Appends a switch with a value that is force quoted. This will quote as if <see cref="CommandLineBuilder.IsQuotingRequired(string)"/>
+        /// returns true for <paramref name="parameter"/>. That means even simple values will be quoted.
+        /// </summary>
+        internal void AppendTextWithForceQuoting(string parameter)
+        {
+            Debug.Assert(!_isQuotingRequired);
+            _isQuotingRequired = true;
+            try
+            {
+                AppendTextWithQuoting(parameter);
+            }
+            finally
+            {
+                _isQuotingRequired = false;
+            }
+        }
+
+        /// <summary>
+        /// Appends a switch with a value that is force quoted. This will quote as if <see cref="CommandLineBuilder.IsQuotingRequired(string)"/>
+        /// returns true for <paramref name="parameter"/>. That means even simple values will be quoted.
+        /// </summary>
+        internal void AppendSwitchForceQuoted(string switchName, string parameter)
+        {
+            AppendSwitch(switchName);
+            _isQuotingRequired = true;
+            try
+            {
+                AppendTextWithQuoting(parameter);
```

Comment:
Despite the method name saying it will append with quoting the implementation is append with maybe quoting. Have to override `IsQuotingRequired` to force quoting to occur here.